src/plugins/mission/mission_impl.cpp

```diff
@@ -467,6 +468,73 @@ MissionImpl::convert_to_int_items(const std::vector<MissionItem>& mission_items)
             int_items.push_back(next_item);
         }
 
+        if (item.vehicle_action != VehicleAction::None) {
```

Comment:
So is this added after the waypoint or before? I'm not quite sure on that yet.

Reply:
I read through this again and I don't think this is correct. So if you say that the first item of a mission would be a takeoff, then the current code would add a `MAV_CMD_NAV_WAYPOINT` at the beginning of this function, and then later down here a `MAV_CMD_NAV_TAKEOFF`, so that would be the wrong way round.

I think the correct way would be to add either a TAKEOFF, LAND, or WAYPOINT as the first item and then add the TRANSITION ones later on, like here. And then for the inverse function this would have to be taken care of in the same way.

Reply:
Current implementation adds vehicle action after any other action (waypoint, gimbal, camera). I would prefer different implementation. I agree, that `MAV_CMD_NAV_TAKEOFF` action should be placed before ` MAV_CMD_NAV_WAYPOINT`. In case of LAND and TRANSITION commands, I would prefer to add them after all other actions. It is more intuitive to place waypoint and then require vehicle to land or do transition.